File: 백준/Bronze/29752. 최장 스트릭/최장 스트릭.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt(); // 일 수
        int[] solvedPerDay = new int[N];

        // 각 날짜별 문제 풀이 횟수 입력
        for (int i = 0; i < N; i++) {
            solvedPerDay[i] = sc.nextInt();
        }

        int maxStreak = 0;
        int currentStreak = 0;

        for (int i = 0; i < N; i++) {
            if (solvedPerDay[i] > 0) {
                currentStreak++;
                maxStreak = Math.max(maxStreak, currentStreak);
            } else {
                currentStreak = 0;
            }
        }

        System.out.println(maxStreak);
    }
}

File: 백준/Bronze/29752. 최장 스트릭/최장 스트릭.py
def longest_streak(N, solved_per_day):
    max_streak = 0
    current_streak = 0

    for i in range(N):
        if solved_per_day[i] > 0:
            current_streak += 1
            max_streak = max(max_streak, current_streak)
        else:
            current_streak = 0

    return max_streak

# 입력 받기
N = int(input().strip())
solved_per_day = list(map(int, input().strip().split()))

# 가장 긴 연속된 풀이 기간 출력
print(longest_streak(N, solved_per_day))